app/src/main/java/com/example/una/CharitySearchListActivity.java

```diff
@@ -42,17 +48,75 @@ protected void onCreate(Bundle savedInstanceState) {
         rvCharities.setAdapter(charitySearchAdapter);
-        rvCharities.setLayoutManager(new LinearLayoutManager(this));
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        rvCharities.setLayoutManager(linearLayoutManager);
         rvCharities.addItemDecoration(new DividerItemDecoration(rvCharities.getContext(), DividerItemDecoration.VERTICAL));
         String query = getIntent().getStringExtra("query");
-        fetchCharities(query);
+
+        scrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
+                // triggered only when new data needs to be appended to the list
+                fetchCharities(query, page);
+            }
+        };
+        // Adds the scroll listener to RecyclerView
+        rvCharities.addOnScrollListener(scrollListener);
+
+        fetchCharities(query, 1);
         setTitle(query);
     }
 
-    private void fetchCharities(String query) {
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // inflate the menu; this adds items to the action bar
+        getMenuInflater().inflate(R.menu.menu_charity_search, menu);
+        final MenuItem searchItem = menu.findItem(R.id.action_search);
+        final SearchView searchView = (SearchView) searchItem.getActionView();
+        // make the search bar cover the title
+        searchView.setMaxWidth(Integer.MAX_VALUE);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                Intent searchListActivity = new Intent(CharitySearchListActivity.this, CharitySearchListActivity.class);
+                searchListActivity.putExtra("query", query);
+                startActivity(searchListActivity);
+
+                // reset SearchView
+                searchView.clearFocus();
+                searchView.setQuery("", false);
+                searchView.setIconified(true);
+                searchItem.collapseActionView();
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String s) {
+                return false;
+            }
+        });
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // handle item selection
+        switch (item.getItemId()) {
+            case R.id.action_search:
+                // handle this selection
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void fetchCharities(String query, int page) {
         RequestParams params = new RequestParams();
         params.put(API_PAGE_SIZE_PARAM, 10);
         params.put(API_SEARCH_PARAM, query);
+        if (page != 1) {
```

Comment:
first page is 1, not 0? Also does the API break if you pass the page number for first page?

Reply:
The API documentation says that the default pageNum value is 1. Should I still be putting 0?

Reply:
in that case one should work. But you have to be are about the scroll listener passing the pagenums correctly.